### ISSUE TITLE
[CLOSED] Site Options Update

### ISSUE THREAD

**Issue by corsacca** [DiscipleTools/disciple-tools-theme#246]
<a href="https://github.com/ChrisChasm"><img src="https://avatars3.githubusercontent.com/u/7318348?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [ChrisChasm](https://github.com/ChrisChasm)**
_Tuesday Sep 26, 2017 at 20:32 GMT_
_Originally opened as https://github.com/DiscipleTools/disciple-tools/pull/166_

----

----

_**[ChrisChasm](https://github.com/ChrisChasm)** included the following code: https://github.com/DiscipleTools/disciple-tools/pull/166/commits_

**Comment by corsacca** [comment]
<a href="https://github.com/ChrisChasm"><img src="https://avatars3.githubusercontent.com/u/7318348?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ChrisChasm](https://github.com/ChrisChasm)**
_Wednesday Sep 27, 2017 at 14:34 GMT_

----

@burntsun Quick explanation (maybe not a great one but here goes :). "Escape Correction" referred to the fact the I was having escape issues with the way I was processing the lists. So because it was the end of the day, I was returning that section back to an "unfinished" but "non-erroring" state, i.e. "escape correction". It would have been better to leave a TODO but I am actively working on it today and tomorrow. I mostly wanted to get the new get_option for site settings to you guys because I thought @corsacca was actively needing that. Otherwise, I'm going to be working in the branch.

Re: I've already removed `dt_print_list_contact_type` and am pursuing a different logic. Here is the reversal. https://github.com/DiscipleTools/disciple-tools/commit/4583d810beaa46e1f25d2a8b63ad9298eec71d6c#diff-7f60639b73a0af29f8f8dff486428f13L260

I tend to write and rewrite rapidly. I tend to be an example of that literary saying, "Every word I've written, I've rewritten." I use the committing less as a delivery for perfectly implemented code, and more as the iterative advancement of writing and rewriting. It's often I'll abandon a development path and go back to previous code, because it ended up being a superior way of executing something. 

Re: The naming conventions. This would be a good discussion. Prefixing is a pattern we are following, so that our functions don't collide with the wider system and other plugins. _get_ is involved in value returns, which is why I didn't include `get` in `dt_list_contact_type` because it was echoing html (even though I've abandoned that. But I think we should formalize some naming conventions further.

**Comment by corsacca** [comment]
<a href="https://github.com/burntsun"><img src="https://avatars3.githubusercontent.com/u/20836899?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [burntsun](https://github.com/burntsun)**
_Wednesday Sep 27, 2017 at 15:23 GMT_

----

> I tend to write and rewrite rapidly. I tend to be an example of that literary saying, "Every word I've written, I've rewritten." I use the committing less as a delivery for perfectly implemented code, and more as the iterative advancement of writing and rewriting. It's often I'll abandon a development path and go back to previous code, because it ended up being a superior way of executing something.

I like this a lot, and I agree with it completely. The better and more mature a product as, the more its code has been rewritten and rewritten. And that's even without considering changing requirements! We are using Git so as to be unafraid to rewrite code if need be. I am not asking you or anyone to only commit perfectly implemented code.

Having said that, my preference is for commits that are well-named, atomic, and that represent a step forwards. If the commit introduces technical debt deliberately, it must be made clear that this was deliberate and temporary. This is especially important when developing in a team, our commits are pieces of communication to other developers. (This paragraph, [On Sausage Making](https://sethrobertson.github.io/GitBestPractices/#sausage), is particularly good read about this topic and summarises differing opinions.)

I'm glad that the commit 4583d81 seems to have fixed the issue. At the time that I made this comment, there was no TODO comment and the commit 4583d81 had not been pushed yet (I think!). So I pointed out that the commit  943451e introduced a bug and needed to be fixed. I also spent some time talking about naming patterns, as a side-note, and I'm glad to hear your points on this in our discussion.
